src/test/scala/generatortools/io/TestModule.scala

```diff
@@ -0,0 +1,106 @@
+// See LICENSE for license details.
+
+package generatortools.io
+
+import chisel3._
+import chisel3.experimental.{ChiselRange, Interval}
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import firrtl.stage.FirrtlSourceAnnotation
+import org.scalatest.{FlatSpec, Matchers}
+
+/** Makes modules with Intervals, custom clocks, etc. compatible with the old testing strategy.
+  * dutFactory = DUT module
+  * useGlobalClk/useGlobalRst = use implicit clock and reset
+  */
+class TestModule[T <: Module](val dutFactory: () => T,
+                              useGlobalClk:   Boolean = true,
+                              useGlobalRst:   Boolean = true,
+                              name:           String = "")
+    extends Module {
+
+  /** Generate standard Chisel top-level IO */
+  private def createTopIO[R <: Record](intIO: R) = {
+    // Internally clones type
+    // new CustomBundle(intIO.elements.toList.map { case (field, elt) => field -> ConvertType(elt) }: _*)
+    intIO.cloneType
+  }
+
+  // Wrap + connect DUT
+  val dut = Module(dutFactory())
+  val io = IO(createTopIO(dut.io))
+
+  if (name.nonEmpty) dut.suggestName(name)
+
+  // Make sure global clock/reset functionality doesn't accidentally get mixed in
+  if (useGlobalClk) dut.clock := clock
+  else dut.clock := false.B.asClock
+  if (useGlobalRst) dut.reset := reset
+  else dut.reset := true.B
+
+  Connect(io, dut.io, if (useGlobalClk) Some(clock) else None)
```

Comment:
```suggestion
io <> dut.io
```

Reply:
Does this work? I remember bulk connect being really wonky in earlier chisel3 days, so maybe this was necessary, but I think `a <> b` should work when one is a `cloneType` of the other, right?

Reply:
Removed for now, we'll see how it evolves in fft-generator